include/stdx/atomic_bitset.hpp

```diff
@@ -159,28 +159,35 @@ class atomic_bitset {
         return *this;
     }
 
-    template <typename T> auto reset(T idx) -> bitset_t {
+    template <typename T>
+    auto reset(T idx, std::memory_order order = std::memory_order_seq_cst)
+        -> bitset_t {
         auto const pos = static_cast<std::size_t>(to_underlying(idx));
-        return bitset_t{
-            atomic::fetch_and(storage, static_cast<elem_t>(~(bit << pos)))};
-    }
-
-    auto reset(std::memory_order order = std::memory_order_seq_cst)
-        LIFETIMEBOUND -> atomic_bitset & {
-        atomic::store(storage, elem_t{}, order);
-        return *this;
+        return bitset_t{atomic::fetch_and(
+            storage, static_cast<elem_t>(~(bit << pos)), order)};
     }
 
     auto
     reset(lsb_t lsb, msb_t msb,
           std::memory_order order = std::memory_order_seq_cst) -> bitset_t {
-        return set(lsb, msb, false, order);
+        auto const l = to_underlying(lsb);
+        auto const m = to_underlying(msb);
+        auto const shifted_value = bit_mask<elem_t>(m, l);
+        return bitset_t{atomic::fetch_and(storage, ~shifted_value, order)};
```

Comment:
question: did you inline `set` here? just curious to know why.

Reply:
a) `set` is more general - we can avoid testing the `value` passed
b) symmetry between `set`, `reset` and `flip`